### PR TITLE
network: add logs for different RelayTx errors

### PR DESCRIPTION
**Pull request by AnnaShaleva**
It's hard to distinguish validation errors and users need more specific data.

### CODE REVIEW

**Review by fyrchik — APPROVED** [review]

File: pkg/network/server.go [hunk 1]
```diff
@@ -1163,30 +1165,30 @@ func (s *Server) relayBlocksLoop() {
 }
 
 // verifyAndPoolTX verifies the TX and adds it to the local mempool.
-func (s *Server) verifyAndPoolTX(t *transaction.Transaction) RelayReason {
+func (s *Server) verifyAndPoolTX(t *transaction.Transaction) (RelayReason, error) {
```
Comment (roman-khimov) [inline pkg/network/server.go:1168]: Return `error` here.

File: pkg/network/server.go [hunk 2]
```diff
@@ -1163,30 +1165,30 @@ func (s *Server) relayBlocksLoop() {
...
 	}
-	return RelaySucceed
+	return RelaySucceed, nil
 }
 
 // RelayTxn a new transaction to the local node and the connected peers.
 // Reference: the method OnRelay in C#: https://github.com/neo-project/neo/blob/master/neo/Network/P2P/LocalNode.cs#L159
-func (s *Server) RelayTxn(t *transaction.Transaction) RelayReason {
-	ret := s.verifyAndPoolTX(t)
+func (s *Server) RelayTxn(t *transaction.Transaction) (RelayReason, error) {
```
Comment (roman-khimov) [inline pkg/network/server.go:1186]: And here.

File: pkg/rpc/server/server.go [hunk 3]
```diff
@@ -1331,7 +1331,8 @@ func (s *Server) sendrawtransaction(reqParams request.Params) (interface{}, *res
 	if err != nil {
 		return nil, response.ErrInvalidParams
 	}
-	return getRelayResult(s.coreServer.RelayTxn(tx), tx.Hash())
+	r, _ := s.coreServer.RelayTxn(tx)
+	return getRelayResult(r, tx.Hash())
```
Comment (roman-khimov) [inline pkg/rpc/server/server.go:1335]: The deal with this error here (in `getRelayResult`), RPC server imports `core` anyway.

File: pkg/network/server.go [hunk 4]
```diff
@@ -152,9 +152,9 @@ func newServerFromConstructors(config ServerConfig, chain blockchainer.Blockchai
 				Log:     log,
 			}
 			n, err := notary.NewNotary(cfg, s.notaryRequestPool, func(tx *transaction.Transaction) error {
-				r := s.RelayTxn(tx)
+				r, err := s.RelayTxn(tx)
 				if r != RelaySucceed {
-					return fmt.Errorf("can't relay completed notary transaction: hash %s, reason: %s", tx.Hash().StringLE(), r.String())
+					return fmt.Errorf("can't relay completed notary transaction: hash %s, reason: %s, error message: %w", tx.Hash().StringLE(), r.String(), err)
```
Comment (roman-khimov) [inline pkg/network/server.go:157]: And just wrap the error here.

File: pkg/network/server.go [hunk 5]
```diff
@@ -185,11 +185,12 @@ func newServerFromConstructors(config ServerConfig, chain blockchainer.Blockchai
 			return nil, fmt.Errorf("can't initialize Oracle module: %w", err)
 		}
 		orc.SetOnTransaction(func(tx *transaction.Transaction) {
-			r := s.RelayTxn(tx)
+			r, err := s.RelayTxn(tx)
 			if r != RelaySucceed {
 				orc.Log.Error("can't pool oracle tx",
 					zap.String("hash", tx.Hash().StringLE()),
-					zap.Uint8("reason", byte(r)))
+					zap.String("reason", r.String()),
```
Comment (roman-khimov) [inline pkg/network/server.go:192]: Log here.

**Review by roman-khimov — CHANGES_REQUESTED** [review]

File: pkg/network/server.go (hunk 1 above)
Comment (AnnaShaleva) [reply pkg/network/server.go]: Fixed, looks better now.

File: pkg/network/server.go [hunk 6]
```diff
@@ -185,11 +183,10 @@ func newServerFromConstructors(config ServerConfig, chain blockchainer.Blockchai
 			return nil, fmt.Errorf("can't initialize Oracle module: %w", err)
 		}
 		orc.SetOnTransaction(func(tx *transaction.Transaction) {
-			r := s.RelayTxn(tx)
-			if r != RelaySucceed {
+			if err := s.RelayTxn(tx); err != nil {
 				orc.Log.Error("can't pool oracle tx",
 					zap.String("hash", tx.Hash().StringLE()),
-					zap.Uint8("reason", byte(r)))
+					zap.String("error", err.Error()))
```
Comment (roman-khimov) [inline pkg/network/server.go:189]: `zap.Error()`?

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**